### PR TITLE
Extend HCS label parsing

### PR DESCRIPTION
**Pull request by ieivanov**
This PR extend HSC site label parsing with the format like `1-Pos000_000`

### CODE REVIEW

**Comment by ziw-liu** [comment]
Can you modify the failing test case to reflect the expected behavior?

**Comment by ieivanov** [comment]
Can I defer this to you? I think in the `MM20_ome-tiffs` collection there are datasets that use the `1-Pos000_000` name format.

**Comment by ieivanov** [comment]
The tests occasionally fail because `test_converter_ometiff_hcs_numerical` samples from `MM20_ome-tiffs` and enforces hcs format (`hcs_plate=True`) even though some datasets in that collection do not have more than one position

File: iohub/convert.py [hunk 1]
```diff
@@ -307,15 +307,15 @@ def _get_pos_names(self):
...
+        try:
             self.pos_names = []
             for p in range(self.p):
                 try:
                     name = self.reader.stage_positions[p]["Label"]
                 except (IndexError, KeyError):
                     name = p
                 self.pos_names.append(name)
-        else:
+        except Exception:
```
Comment (ziw-liu) [inline iohub/convert.py:318]: Can you enumerate the expected exceptions here?

Comment (ziw-liu) [reply iohub/convert.py:318]: Also what happens if the dataset has multiple positions but gets caught in this branch? Should it `raise`?

Comment (ieivanov) [reply iohub/convert.py]: The confusing part seems to be the nested `try` statements, which are not necessary. Simplified.

**Comment by ieivanov** [comment]
@edyoshikun @ziw-liu let's try to merge this today, we'll likely need to use this branch to reconstruct the data from tomorrow's experiment. This branch will also fix the failing tests in the main branch.

**Review by ziw-liu — APPROVED** [review]
@edyoshikun can you take a look at the proposed behavior and confirm it does not break your workflow?

**Review by edyoshikun — APPROVED** [review]
This does not break my workflow since we still support the original '0/0/0' position. It can also adapt to this

**Pull request merged**